gems/activerecord/7.2/activerecord-7.2.rbs

```diff
@@ -46,6 +39,11 @@ module ActiveRecord
     extend RuntimeRegistry
   end
 
+  module QueryMethods
+    def with_recursive: (*untyped) -> untyped
+    def with_recursive!: (*untyped) -> untyped
+  end
+
   module QueryMethods
```

Comment:
How about merging `module QueryMethods`?

Reply:
I'm worried about the sub-modules and classes becoming bigger.

```
module QueryMethods
  class WhereChain[Relation]
     (long long definitions)
   end

  module OtherMod
    (also very long)
  end

  def with_recursive: (*untyped) -> untyped
  def with_recursive!: (*untyped) -> untyped
end
```

I know I worry too much, but activerecord is a large library.
Therefore, I think it's better to separate definitions per module or class.
For example, `ActiveRecord::Relation` and `ActiveRecord::Associations` have many sub-modules.